### ISSUE TITLE
1.7.0 invoke-IcingaCheckMemory strange results

### ISSUE THREAD

**Issue by Laszlo-Fiat** [Icinga/icinga-powershell-plugins#245]
I've upgraded from 1.3.1 to 1.7.0 on one of the servers, since then I get a Warning from Memory Check. I was playing around with the examples, but cannot get an OK state.

PS C:\Windows\system32> icinga
******************************************
**      Icinga for Windows v1.7.0       **
** Copyright (c) 2021 Icinga GmbH | MIT **
** User environment domain\user  **
******************************************
icinga> Invoke-IcingaCheckMemory
[OK] Memory Usage
| 'memory_percent_used'=47.121627%;;;0;100 'used_bytes'=7539196000B;;;0;15999440000
0

Issued as in the documentation here: https://icinga.com/docs/icinga-for-windows/latest/plugins/doc/plugins/09-Invoke-IcingaCheckMemory/
icinga> Invoke-IcingaCheckMemory -Verbosity 3 -Warning 60 -Critical 80
[UNKNOWN] Memory Usage [UNKNOWN] Memory Percent Used (45.629863%) [CRITICAL] Used Bytes (7.30GB) (All must be [OK])
\_ [UNKNOWN] Memory Percent Used: Invalid range specified for threshold: InputValue "45,629863248219487253338394340" and
 Threshold 3.49258206118655E-07
\_ [CRITICAL] Used Bytes: 7.30GB is greater than threshold 0.00000008B
| 'memory_percent_used'=45.629863%;3.49258206118655E-07;;0;100 'used_bytes'=7300522000B;0.000000;0.000000;0;15999440000

3

It looks like the Warning and Critical levels are not handled correctly.

icinga> Invoke-IcingaCheckMemory -Verbosity 3 -WarningPercent 80 -CriticalPercent 90
[CRITICAL] Memory Usage [CRITICAL] Used Bytes (8.35GB) (All must be [OK])
\_ [OK] Memory Percent Used: 52.173920%
\_ [CRITICAL] Used Bytes: 8.35GB is greater than threshold 14.3994951248169B
| 'memory_percent_used'=52.173920%;80;90;0;100 'used_bytes'=8347534000B;12.799551;14.399495;0;15999440000
2

This makes the Percent to give OK, and there seems to be a calculation to return 12.79 and 14.39, which would be ok for GB, but they seems to be used as bytes.

**Comment by puseidr** [comment]
just add "%" to the warning and critical threshold like: Invoke-IcingaCheckMemory -Verbosity 3 -Warning 60% -Critical 80%

you can also add "GB": Invoke-IcingaCheckMemory -Verbosity 3 -Warning 12GB -Critical 15GB

**Comment by Laszlo-Fiat** [comment]
Hello,

I've just tried this: 
PS C:\Windows\system32> icinga
******************************************
**      Icinga for Windows v1.7.0       **
** Copyright (c) 2021 Icinga GmbH | MIT **
** User environment domain\admin  **
******************************************
icinga> Invoke-IcingaCheckMemory -Verbosity 3 -Warning 80% -Critical 90%
[UNKNOWN]: Icinga Invalid Input Error was thrown: ConversionUnitMissing

Unable to parse input value. You have to add an unit to your input value. Example: "10GB". Allowed units are: "B, KB, MB
, GB, TB, PB, KiB, MiB, GiB, TiB, PiB".

**However I may not have the latest plugins**, even though

PS C:\Windows\system32> use-icinga
PS C:\Windows\system32> update-icinga
[Notice]: No update package found for component "agent"
[Notice]: No update package found for component "plugins"
[Notice]: No update package found for component "framework"

**Is there versioning in the plugins what I can check?**

Thank you!

**Comment by LordHepipud** [comment]
Hello,

you can use

```powershell
Show-Icinga
```

to receive a full list of the current Icinga for Windows environment. This should point to which plugin version is installed right now.

**Comment by Laszlo-Fiat** [comment]
PS C:\Windows\system32> use-icinga
PS C:\Windows\system32> Show-Icinga
Icinga for Windows environment
-----------

Installed components on this system

Component    Version   Available
---          ---       ---
agent        2.12.3
plugins      1.3.1
framework    1.7.0

Available versions flagged with "*" mean that this component is locked to this version

Environment configuration

PowerShell Root                 => C:\Program Files\WindowsPowerShell\Modules\
Icinga for Windows Service Path => C:\Program Files\icinga-framework-service
Icinga for Windows Service User => NT Authority\NetworkService
Icinga Agent Path               => C:\Program Files\ICINGA2\
Icinga Agent User               => NT AUTHORITY\NetworkService
Defined Default User            =>
Icinga Managed User             => False
PowerShell Version              => 5.1.14409.1018
Operating System                => Microsoft Windows Server 2012 R2 Standard
Operating System Version        => 6.3.9600
JEA Context                     =>
JEA Session File                =>

Icinga for Windows Certificate

Not installed

List of configured repositories on this system. The list order matches the apply order.

No repositories configured

**Comment by LordHepipud** [comment]
Okay, for some reason plugins in version 1.3.1 are still installed. Are you sure you upgraded them? Is there any other PowerShell module directoy on which you installed the plugins to?

**Comment by Laszlo-Fiat** [comment]
The issue is that I haven't configured any repositories, so it couldn't find any updates. 
I have added the stable repository since, but it seems it doesn't want to go through the proxy that is mandatory to use:

PS C:\Windows\system32> show-icinga
[Error]: Unable to resolve repository URL "https://packages.icinga.com/IcingaForWindows/stable/ifw.repo.json" for reposi
tory "Icinga Stable": The request was aborted: Could not create SSL/TLS secure channel.
[Error]: Unable to resolve repository URL "https://packages.icinga.com/IcingaForWindows/stable/ifw.repo.json" for reposi
tory "Icinga Stable": The request was aborted: Could not create SSL/TLS secure channel.
[Error]: Unable to resolve repository URL "https://packages.icinga.com/IcingaForWindows/stable/ifw.repo.json" for reposi
tory "Icinga Stable": The request was aborted: Could not create SSL/TLS secure channel.
Icinga for Windows environment
-----------

Installed components on this system

Component    Version   Available
---          ---       ---
agent        2.12.3
plugins      1.3.1
framework    1.7.0

So basically we know now that I have old plugins, so this ticket can be closed. Thank you for your help, and sorry for not RTFM first. I did RTFM when installed Framework 1.3.1 and Plugins 1.3.1, but didn't go deep enough during the upgrade.
